### PR TITLE
Feature/jc 703 initialize react app

### PR DESCRIPTION
**Pull request by fongradnastya**
Initialized a react app and created a basic page layout

### CODE REVIEW

File: apps/react/src/features/genres/components/GenreDetails/GenreDetails.module.css [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.main {
+  flex-grow: 1;
+  padding: var(--space-20) var(--space-6);
+  transition: margin 225ms cubic-bezier(0.4, 0, 0.6, 1);
+  margin-left: -280px;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreDetails/GenreDetails.module.css:5]: Why do you need such a large margin? Put the value in a separate variable

File: apps/react/src/features/genres/components/GenreDetails/GenreDetails.module.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+.mainOpen {
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreDetails/GenreDetails.module.css:8]: You aren't using this class

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+.form {
+  padding: 2px 4px;
+  display: flex;
+  align-items: center;
+  width: 100%;
+}
+
+.inputBase {
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css:14]: Use BEM

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  margin-left: 1rem;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css:15]: We definitely need to use `rem` here, not `px`.

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+				aria-label="search"
+				color="primary"
+			>
+				<SearchIcon />
+			</IconButton>
+		</Paper>
+	</Box>
+);
+
+/** Memoized version of GenreFiltersComponent for performance optimization. */
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreFilters/GenreFilters.tsx:37]:
```suggestion
/** Genre filters component. */
```

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import { memo, FC } from 'react';
+import Paper from '@mui/material/Paper';
+import InputBase from '@mui/material/InputBase';
+import IconButton from '@mui/material/IconButton';
+import SearchIcon from '@mui/icons-material/Search';
+import { Box, Typography } from '@mui/material';
+
+import styles from './GenreFilters.module.css';
+
+/** Filters for anime genres. */
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreFilters/GenreFilters.tsx:10]: No comments are needed here

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+			Filters
+		</Typography>
+		<Paper
+			component="form"
+			className={styles.form}
+		>
+			<InputBase
+				className={styles.inputBase}
+				placeholder={ `Search Genres ...` }
+				inputProps={{ 'aria-label': 'search google maps' }}
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreFilters/GenreFilters.tsx:23]:
```suggestion
				placeholder='Search Genres ...'
				aria-label='search google maps'
```

File: apps/react/src/features/genres/components/GenreLayout/GenreLayout.module.css [hunk 8]
```diff
@@ -0,0 +1,28 @@
+.main {
+  display: flex;
+  gap: var(--space-8);
+  flex-grow: 1;
+  padding: 100px var(--space-6) 0 var(--space-6);
+  transition: margin 225ms cubic-bezier(0.4, 0, 0.6, 1);
+  margin-left: calc(-1 * var(--sidebar-width));
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreLayout/GenreLayout.module.css:7]: Why is `-1` here?

File: apps/react/src/features/genres/components/GenreLayout/GenreLayout.module.css [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  flex-shrink: 0;
+}
+
+.button {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  margin-left: 600px;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreLayout/GenreLayout.module.css:27]: Why is there such a large margin here?

File: apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+	const navigate = useNavigate();
+	const { genreId } = useParams<{ genreId: string; }>();
+
+	useEffect(() => {
+		dispatch(fetchGenres());
+	}, [dispatch]);
+
+	if (isLoading) {
+		return <div>Loading</div>;
+	}
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx:30]: Let's add a ternary operator below

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+import { GenreFilters } from '../GenreFilters';
+
+import styles from './GenreList.module.css';
+
+type GenresListProps = {
+
+	/** An array of anime genres. */
+	readonly genres: Array<AnimeGenre>;
+
+	/** Click handler. */
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:17]: Now it isn't clear from the comment what the function does

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+import { GenreListItem } from '../GenreListItem';
+import { GenreFilters } from '../GenreFilters';
+
+import styles from './GenreList.module.css';
+
+type GenresListProps = {
+
+	/** An array of anime genres. */
+	readonly genres: Array<AnimeGenre>;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:15]:
```suggestion
	readonly genres: readonly AnimeGenre[];
```

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+	/** Click handler. */
+	onGenreClick: (id: number) => void;
+};
+
+/** A list of anime genres. */
+const GenresListComponent: FC<GenresListProps> = ({ genres, onGenreClick }: GenresListProps) => {
+	const { genreId } = useParams<{ genreId: string; }>();
+	const [selectedGenreId, setSelectedGenreId] = useState<number | undefined>(genreId ? +genreId : undefined);
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:24]:
```suggestion
	const [selectedGenreId, setSelectedGenreId] = useState<number | undefined>(genreId ? Number(genreId) : undefined);
```
https://wiki.saritasa.rocks/frontend/languages/ts-js/best-practices#dont-use-implicit-type-casting

File: apps/react/src/features/genres/pages/GenrePage/GenrePage.module.css [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+.none {
+  display: none;
+}
+
+.drawer {
+  width: var(--sidebar-width);
+  flex-shrink: 0;
+}
+
+.drawerPaper {
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/pages/GenrePage/GenrePage.module.css:40]: This class isn't used

File: package.json [hunk 15]
```diff
@@ -37,6 +37,11 @@
     "@angular/platform-browser": "18.0.5",
     "@angular/platform-browser-dynamic": "18.0.5",
     "@angular/router": "18.0.5",
+    "@emotion/react": "^11.13.0",
+    "@emotion/styled": "^11.13.0",
+    "@fontsource/roboto": "^5.0.14",
```
Comment (SofyaKirillova) [inline package.json:42]: It seems these libraries aren't being used

**Review by SofyaKirillova — CHANGES_REQUESTED** [review]

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css [hunk 16]
```diff
@@ -4,18 +4,18 @@
   margin: 0 var(--space-5) var(--space-5) 0;
 }
 
-.form {
+.filters__form {
   padding: 2px 4px;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css:8]: Please use variables

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 17]
```diff
@@ -12,29 +12,32 @@ import styles from './GenreList.module.css';
 type GenresListProps = {
 
 	/** An array of anime genres. */
-	readonly genres: Array<AnimeGenre>;
+	readonly genres: readonly AnimeGenre[];
 
-	/** Click handler. */
+	/** Handles displaying genre details on click. */
 	onGenreClick: (id: number) => void;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:18]:
```suggestion
	readonly onGenreClick: (id: number) => void;
```

File: apps/react/src/features/genres/components/GenreListItem/GenreListItem.tsx [hunk 18]
```diff
@@ -14,7 +14,6 @@ type GenreListItemProps = {
 	readonly onClick: () => void;
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/GenreListItem/GenreListItem.tsx:14]: Now it is unclear by name what the function does

File: apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+	{ name: 'Anime', path: '/anime' },
+	{ name: 'Genres', path: '/genres' },
+	{ name: 'Studios', path: '/studios' },
+];
+
+const loginRoutes = [
+	{ name: 'Login', path: '/login' },
+	{ name: 'Logout', path: '/logout' },
+	{ name: 'Profile', path: '/profile' },
+];
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx:34]: Let's add a type

File: apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+
+const GenrePageComponent: FC = () => {
+	const open = useSelector(selectIsDrawerOpen);
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx:37]:
```suggestion
	const drawerOpe = useSelector(selectIsDrawerOpen);
```

File: apps/react/src/features/genres/components/NavigationList/NavigationList.tsx [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import InboxIcon from '@mui/icons-material/MoveToInbox';
+import MailIcon from '@mui/icons-material/Mail';
+
+type NavigationListProps = {
+
+	/** An array of navigation objects. */
+	items: {
```
Comment (SofyaKirillova) [inline apps/react/src/features/genres/components/NavigationList/NavigationList.tsx:13]:
```suggestion
	readonly items: readonly {
```

**Review by SofyaKirillova — APPROVED** [review]

File: apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css (hunk 3 above)
Comment (timofei-maximov-saritasa) [reply apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css]: @fongradnastya @SofyaKirillova We don't use BEM in react

Comment (timofei-maximov-saritasa) [reply apps/react/src/features/genres/components/GenreFilters/GenreFilters.module.css]: @fongradnastya If you changed it to BEM, leave it as-is but there is no point in BEM with CSS modules:
1. We use components(BEM existed before this concept, so it doesn't help us separate blocks)
2. It's easier to use camelCase in classNames - `className={styles.fooBar}` vs `className={styles['foo__bar']}`

File: apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+import { Box, ListItemText, IconButton } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import { useNavigate, Outlet, useParams } from 'react-router-dom';
+
+import { GenresList } from '../GenreList';
+
+import styles from './GenreLayout.module.css';
+
+const GenreLayoutComponent: FC = () => {
+	const open = useSelector(selectIsDrawerOpen);
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx:16]:
```suggestion
	const isDrawerOpen= useSelector(selectIsDrawerOpen);
```

File: apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+		dispatch(fetchGenres());
+	}, [dispatch]);
+
+	if (isLoading) {
+		return <div>Loading</div>;
+	}
+
+	const handleGenreClick = (id: number) => {
+		navigate(`/genre/${id}`);
+	};
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx:33]: Let's group the return statement at the bottom of the component

Comment (andrey-gazzaev) [reply apps/react/src/features/genres/components/GenreLayout/GenreLayout.tsx:33]:
```
// states...

// hadlers...

// effects...

// some code...
  return <loading/>
//  some code...
  return <YourComponent/>

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+import { GenreFilters } from '../GenreFilters';
+
+import styles from './GenreList.module.css';
+
+type GenresListProps = {
+
+	/** An array of anime genres. */
+	readonly genres: readonly AnimeGenre[];
+
+	/** Handles displaying genre details on click. */
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:17]:
```suggestion
	/** Displaying genre details on click handler. */
```

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 25]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+const GenresListComponent: FC<GenresListProps> = ({ genres, onGenreClick }: GenresListProps) => {
+	const { genreId } = useParams<{ genreId: string; }>();
+	const [selectedGenreId, setSelectedGenreId] = useState<number | undefined>(genreId ? Number(genreId) : undefined);
+
+	const handleGenreClick = (id: number) => {
+		setSelectedGenreId(id);
+		onGenreClick(id);
+	};
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:28]: Use useCallback. Systematic use of this hook will allow you to have a reduced number of rerenders

File: apps/react/src/features/genres/components/NavigationList/NavigationList.tsx [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import InboxIcon from '@mui/icons-material/MoveToInbox';
+import MailIcon from '@mui/icons-material/Mail';
+
+import { NavigationProps } from '../../utils/navigationProps';
+
+type NavigationListProps = {
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/components/NavigationList/NavigationList.tsx:12]:
```suggestion
type Props = {
```
If you are not exporting a type, you can use the usual form of the name

File: apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+];
+
+const loginRoutes: NavigationProps[] = [
+	{ name: 'Login', path: '/login' },
+	{ name: 'Logout', path: '/logout' },
+	{ name: 'Profile', path: '/profile' },
+];
+
+const GenrePageComponent: FC = () => {
+	const iDrawerOpen = useSelector(selectIsDrawerOpen);
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx:38]:
```suggestion
	const isDrawerOpen = useSelector(selectIsDrawerOpen);
```

File: apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+		dispatch(setOpen(true));
+	};
+
+	const handleDrawerClose = () => {
+		dispatch(setOpen(false));
+	};
+	return (
+		<Box className={styles.main}>
+			<CssBaseline />
+			<MuiAppBar className={`${styles['main__app-bar']} ${iDrawerOpen ? styles['main__app-bar_open'] : ''}`}>
```
Comment (andrey-gazzaev) [inline apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx:51]:
```suggestion
			<MuiAppBar className={clsx(styles['main__app-bar'], isDrawerOpen && styles['main__app-bar_open'] )}>
```

https://www.npmjs.com/package/clsx

Comment (andrey-gazzaev) [reply apps/react/src/features/genres/pages/GenrePage/GenrePage.tsx:51]: Try using this library. It will be easier to work with styles

**Review by andrey-gazzaev — APPROVED** [review]

File: apps/react/src/features/genres/components/GenreListItem/GenreListItem.tsx [hunk 29]
```diff
@@ -0,0 +1,32 @@
...
+	/** Anime genre. */
+	readonly genre: AnimeGenre;
+
+	/** Displaying genre details on click handler. */
+	readonly onClick: () => void;
+};
+
+const GenreListItemComponent: FC<Props> = ({ selected, genre, onClick }: Props) => (
+	<ListItem disablePadding>
+		<ListItemButton
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/genres/components/GenreListItem/GenreListItem.tsx:19]: Let's use link component here over button.

`<ListItemButton component={NavLink} to={genreUrl}`

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+const GenresListComponent: FC<Props> = ({ genres, onGenreClick }: Props) => {
+	const { genreId } = useParams<{ genreId: string; }>();
+	const [selectedGenreId, setSelectedGenreId] = useState<number | undefined>(genreId ? Number(genreId) : undefined);
+
+	const handleGenreClick = useCallback((id: number) => {
+		setSelectedGenreId(id);
+		onGenreClick(id);
+	}, [onGenreClick]);
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/features/genres/components/GenreList/GenreList.tsx:28]: You don't need it if you will handle navigation using links

File: apps/react/src/store/drawer/slice.ts [hunk 31]
```diff
@@ -0,0 +1,19 @@
...
+		setOpen(state, action: PayloadAction<boolean>) {
+			state.open = action.payload;
+		},
+	},
+});
+
+/** Sets new open state. */
+export const { setOpen } = drawerSlice.actions;
+
+export default drawerSlice.reducer;
```
Comment (timofei-maximov-saritasa) [inline apps/react/src/store/drawer/slice.ts:19]: Do you need this?

File: apps/react/src/features/genres/components/GenreList/GenreList.tsx (hunk 30 above)
Comment (fongradnastya) [reply apps/react/src/features/genres/components/GenreList/GenreList.tsx]: I started using navigation links but I still need a handler to set the selected genre

**Review by dbitkovski — APPROVED** [review]

File: apps/react/src/store/drawer/slice.ts (hunk 31 above)
Comment (fongradnastya) [reply apps/react/src/store/drawer/slice.ts:19]: I think so, because I need to share the state of the drawer between different pages.

**Pull request merged**